connaisseur/alert.py

```diff
@@ -0,0 +1,177 @@
+    """
+
+    connaisseur_pod_id: str
+    cluster: str
+    timestamp: int
+
+    request_id: str
+    images: list
+
+    template: str
+    receiver_url: str
+    payload: dict
+    headers: dict
+
+    alert_message: str
+    priority: int
+
+    def __init__(self, alert_message, receiver_config, admission_request):
+        if "receiver_url" not in receiver_config or "template" not in receiver_config:
+            raise ConfigurationError(
+                "Either 'receiver_url' or 'template' or both are missing to construct the alert. "
+                "Both can be configured in the 'values.yaml' file in the 'helm' directory"
+            )
+        self.context = {
+            "alert_message": alert_message,
+            "priority": str(receiver_config.get("priority", 3)),
+            "connaisseur_pod_id": os.getenv("POD_NAME"),
+            "cluster": load_config()["cluster_name"],
+            "timestamp": datetime.now(),
+            "request_id": admission_request.get("request", {}).get(
+                "uid", "No given UID"
+            ),
+            "images": (str(get_images(admission_request)) or "No given images"),
+        }
+        self.admission_request = admission_request
+        self.receiver_url = receiver_config["receiver_url"]
+        self.template = receiver_config["template"]
+        self.throw_if_alert_sending_fails = receiver_config.get(
+            "fail_if_alert_sending_fails", True
+        )
+        self.payload = self._construct_payload(receiver_config)
+        self.headers = get_headers(receiver_config)
+
+    def _construct_payload(self, receiver_config):
+        try:
+            with open(
+                "{}/{}.json".format(os.getenv("ALERT_CONFIG_DIR"), self.template),
+                "r",
+            ) as templatefile:
+                template = json.load(templatefile)
+        except Exception as err:
+            raise ConfigurationError(
+                "Template file for alerting payload is either missing or invalid JSON: {}".format(
+                    str(err)
+                )
+            )
+        payload = sanitize(template, self.context)
+        if receiver_config.get("payload_fields") is not None:
+            payload.update(receiver_config.get("payload_fields"))
+        return payload
+
+    def send_alert(self):
+        try:
+            response = requests.post(
+                self.receiver_url, data=self.payload, headers=self.headers
+            )
+            response.raise_for_status()
+            logging.info("sent alert to %s", self.template.split("_")[0])
+        except Exception as err:
+            if self.throw_if_alert_sending_fails:
+                raise AlertSendingError(str(err))
+            logging.error(err)
+        return response
+
+
+def load_config():
+    with open(
+        "{}/alertconfig.json".format(os.getenv("ALERT_CONFIG_DIR")), "r"
+    ) as configfile:
+        return json.loads(configfile.read())
```

Comment:
```suggestion
        return json.load(configfile)
```
`loads` it for strings, `load` for files